iceoryx_posh/source/roudi/memory/memory_provider.cpp

```diff
@@ -86,14 +86,19 @@ cxx::expected<MemoryProviderError> MemoryProvider::create() noexcept
 
     m_memory = memoryResult.value();
     m_size = totalSize;
-    m_segmentId = rp::BaseRelativePointer::registerPtr(m_memory, m_size);
+    auto maybeSegmentId = rp::BaseRelativePointer::registerPtr(m_memory, m_size);
+
+    if (maybeSegmentId.has_value())
```

Comment:
If it does not work it can be considered fatal and we would actually not be able to continue. This cannot really happen for a reasonable number of segments though.

We should generate a fatal error in this case, or at least have a placeholder terminate and a todo for it (will add some `IOX_PANIC` or similar once available) .